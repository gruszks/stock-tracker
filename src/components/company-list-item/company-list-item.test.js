import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import CompanyListItem from './company-list-item';

it('renders correctly', () => {
  const company = {
    change: 1.12,
    changePercent: 0.589,
    currency: 'USD',
    domain: 'facebook.com',
    image: 'https://logo.clearbit.com/facebook.com',
    marketClose: '16:00',
    marketOpen: '09:30',
    name: 'Facebook Inc.',
    price: 189.04,
    region: 'United States',
    symbol: 'FB',
    timezone: 'UTC-04',
    tradingDay: '2019-08-09',
  };
  const { asFragment } = render(
    <CompanyListItem data={company} onDeleteCompany={jest.fn()} />
  );

  expect(asFragment()).toMatchSnapshot();
});
