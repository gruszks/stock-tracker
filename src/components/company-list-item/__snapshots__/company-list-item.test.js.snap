// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should renders correctly 1`] = `
<DocumentFragment>
  <li
    class="company-list-itemstyles__Item-sc-1snnccw-6 hXRuc"
  >
    <div
      class="company-list-itemstyles__Content-sc-1snnccw-3 dgUIOo"
    >
      <div>
        <h3
          class="company-list-itemstyles__Name-sc-1snnccw-2 doaeAC"
        >
          Facebook Inc.
        </h3>
        <span>
          FB
        </span>
         
        <span>
          facebook.com
        </span>
      </div>
      <div>
        <span>
          United States
        </span>
        <span>
          09:30 - 16:00
        </span>
        <span>
          UTC-04
        </span>
      </div>
      <div>
        <span
          title="189.04"
        >
          <strong>
            189.04
          </strong>
        </span>
        <span>
          USD
        </span>
        <span
          class="company-list-itemstyles__State-sc-1snnccw-5 dDPulx"
        >
          <span
            title="1.12"
          >
            1.12
          </span>
          <span
            title="0.589%"
          >
            (0.59%) 
            <img
              alt="up"
              src="up.png"
            />
          </span>
        </span>
        <span>
          Closed: 2019-08-09
        </span>
      </div>
    </div>
    <img
      alt="Facebook Inc."
      class="company-list-itemstyles__Image-sc-1snnccw-0 dOLemJ"
      src="https://logo.clearbit.com/facebook.com"
    />
    <button
      class="company-list-itemstyles__Delete-sc-1snnccw-4 hitWzm"
    >
      <img
        alt="Delete"
        src="delete.png"
      />
    </button>
  </li>
</DocumentFragment>
`;
